server/tests/SubscriptionTreeSpec.hs

```diff
@@ -69,3 +72,44 @@ spec = do
         conn_id = 1 :: Int
       unsubscribe path conn_id (empty :: SubscriptionTree Int String)`shouldBe` empty
 
+    do
+      let
+        conn1    , conn2    , conn3    , conn4 :: Int
+        conn1 = 1; conn2 = 2; conn3 = 3; conn4 = 4
+
+        root = SubscriptionTree (HM.fromList [(conn1, conn1)])
+                                (HM.fromList [ ("foo", root_foo)
+                                             , ("baz", root_baz) ])
+        root_foo = SubscriptionTree (HM.fromList [(conn2, conn2)])
+                                    (HM.fromList [("bar", root_foo_bar)])
+        root_foo_bar = SubscriptionTree (HM.fromList [(conn3, conn3)]) HM.empty
+        root_baz = SubscriptionTree (HM.fromList [(conn4, conn4)]) HM.empty
+
+        value = AE.object ["foo" AE..= value_foo, "baz" AE..= value_baz]
+        value_foo = AE.object ["bar" AE..= value_foo_bar]
+        value_foo_bar = AE.Null
+        value_baz = AE.object []
+
+        broadcast'' path = sortOn fst . execWriter $ broadcast' path value root
+
+      it "notifies everyone on root updates" $ do
+        broadcast'' []
+          `shouldBe` [ (conn1, value)
+                     , (conn2, value_foo)
+                     , (conn3, value_foo_bar)
+                     , (conn4, value_baz)
+                     ]
+
+      it "notifies parents and children about updates" $ do
+        broadcast'' ["foo"]
+          `shouldBe` [ (conn1, value)
+                     , (conn2, value_foo)
+                     , (conn3, value_foo_bar)
+                     ]
+
+      it "notifies parents and children about updates" $ do
+        broadcast'' ["foo", "bar"]
+          `shouldBe` [ (conn1, value)
+                     , (conn2, value_foo)
+                     , (conn3, value_foo_bar)
+                     ]
```

Comment:
Please add  a test that shows that a client subscribed to a subtree will *only* receive that subtree on a change and not the whole tree.

Reply:
The tests catch that; if the clients would receive the whole tree then the test will fail when checking equality with `value_foo` and `value_foo_bar`.

Reply:
Ok. Is there also a test that makes sure that a client does *not* get updated if an unrelated part of the tree is updated?

Reply:
Yes; `conn4` is not present in all of the test examples.